language/stdlib/modules/LibraAccount.move

```diff
@@ -521,43 +540,66 @@ module LibraAccount {
         destroy_signer(new_account);
     }
 
-    /// Create an account with the AssocRoot role at `new_account_address` with authentication key
-    /// `auth_key_prefix` | `new_account_address`
+    /// Create an account for the on-chain config account at
+    /// `CoreAddresses::DEFAULT_CONFIG_ADDRESS()` with authentication key
+    /// `auth_key_prefix` | `new_account_address`. Called in genesis.
     // TODO: can we get rid of this? the main thing this does is create an account without an
     // EventGenerator resource (which is just needed to avoid circular dep issues in gensis)
-    public fun create_genesis_account<Token>(
+    public fun create_config_account<Token>(
+        association_root_role: &RoleId,
+        _on_chain_config_creation_capability: &Capability<CreateOnChainConfig>,
         new_account_address: address,
         auth_key_prefix: vector<u8>
+    ) acquires AccountOperationsCapability {
+        assert(LibraTimestamp::is_genesis(), 0);
+        assert(new_account_address == CoreAddresses::DEFAULT_CONFIG_ADDRESS(), 1);
+        let new_account = create_signer(new_account_address);
+        let role = Roles::new_role(
+            association_root_role,
+            &borrow_global<AccountOperationsCapability>(CoreAddresses::ASSOCIATION_ROOT_ADDRESS()).role_creation_cap,
+            &new_account,
+            Roles::PARENT_VASP_ROLE_ID(),
+        );
+        make_account<Token>(new_account, auth_key_prefix, false, role)
+    }
+
+    /// Creates the root association account in genesis. The role is passed
+    /// in because root association privileges are needed before this
+    /// account can be created.
+    public fun create_root_association_account<Token>(
+        new_account_address: address,
+        auth_key_prefix: vector<u8>,
+        role: RoleId,
     ) {
         assert(LibraTimestamp::is_genesis(), 0);
+        assert(new_account_address == CoreAddresses::ASSOCIATION_ROOT_ADDRESS(), 0);
+        assert(Roles::role_address(&role) == CoreAddresses::ASSOCIATION_ROOT_ADDRESS(), 1);
         let new_account = create_signer(new_account_address);
-        let role_id = 0;
-        make_account<Token>(new_account, auth_key_prefix, false, role_id)
+        make_account<Token>(new_account, auth_key_prefix, false, role)
     }
 
     /// Create a treasury/compliance account at `new_account_address` with authentication key
     /// `auth_key_prefix` | `new_account_address`
     public fun create_treasury_compliance_account<Token>(
-        association: &signer,
+        _tc_creation_capability: &Capability<CreateTreasuryCompliance>,
+        sliding_nonce_creation_capability: &Capability<CreateSlidingNonce>,
```

Comment:
Interesting: if we only want sliding nonces to exist in freshly created TreasuryCompliance account, should we combine the capabilities (i.e., `publishing_sliding_nonce` asks for a `CreateTreasuryCompliance`)?

Reply:
Good question. I'm fine combining these (because I'm not sure about who should be able to have a sliding nonce). My thinking was to keep the privileges as fine-grained as possible, and let the roles define what set of privileges are allowed. But happy to combine these two :)